src/lib/indexer/eventProcessor.ts

```diff
@@ -238,6 +250,47 @@ const DefaultEventProcessors: EventProcessors = {
       ]),
     }
   },
+
+  'match2-cancel': (version, _transaction, _sender, inputs, _outputs) => {
+    if (version !== 1) {
+      throw new Error(`Incompatible version ${version} for match2-cancel process`)
+    }
+
+    const demandALocalId = inputs[0].localId
+    const demandAId = _outputs[0].id
+    const demandBLocalId = inputs[1].localId
+    const demandBId = _outputs[1].id
+    const matchLocalId = inputs[2].localId
+    const matchId = _outputs[2].id
+    const match2Cancel = _outputs[2]
+
+    const attachment: AttachmentRecord = {
+      type: 'insert',
+      id: UUIDv4(),
+      ipfs_hash: getOrError(match2Cancel.metadata, 'comment'),
+    }
+
+    const comment: Match2CommentRecord = {
+      type: 'insert',
+      id: UUIDv4(),
+      state: 'created',
+      match2: matchLocalId,
+      owner: _sender,
+      attachment: attachment.id,
+    }
+
+    return {
+      demands: new Map([
+        [demandALocalId, { type: 'update', id: demandALocalId, latest_token_id: demandAId, state: 'cancelled' }],
+        [demandBLocalId, { type: 'update', id: demandBLocalId, latest_token_id: demandBId, state: 'cancelled' }],
+      ]),
+      attachments: new Map([[attachment.id, attachment]]),
+      match2Comments: new Map([[comment.id, comment]]),
+      matches: new Map([
+        [matchLocalId, { type: 'update', id: matchLocalId, latest_token_id: matchId, state: 'cancelled' }],
+      ]),
+    }
+  },
```

Comment:
This handles the flow where the transaction was submitted by another persona and is therefore a new insert to us. We also need to handle the case where `transaction` is not null

Reply:
I just reverted back what pretty much was before https://github.com/digicatapult/dscp-matchmaker-api/blob/f2d8514f3aaf2f15bc983921c154b5ab8e677f6d/src/lib/indexer/eventProcessor.ts#L263-L284

Reply:
updated handler `match2-cancel` and marking as resolved as this is no longer relevant and was indeed invalid.